### PR TITLE
ops: add stablehlo.fft lowering

### PR DESCRIPTION
**Pull request by dlwh**
Summary
- add stablehlo.fft lowering via MPSGraph FFT APIs
- add fft_ops source to CMake build
- add misc config coverage for jnp.fft.fft
- include explicit parser support for stablehlo.fft to keep this PR self-contained

Validation
- uv pip install -e .
- uv run pytest -q tests/test_ops.py -k "misc and not unsupported"

### CODE REVIEW

File: tests/configs/misc.py [hunk 1]
```diff
@@ -10,5 +11,13 @@ def make_misc_op_configs():
             OperationTestConfig(
                 lambda x: x,
                 numpy.random.standard_normal((4, 5, 6, 8)).transpose((2, 0, 1, 3)),
-            )
+            ),
+            OperationTestConfig(
```
Comment (tillahoffmann) [inline tests/configs/misc.py:16]: Thank you for the contribution! Can we add a few more scenarios here, e.g., multidimensional, ifft, rfft, irfft, explicit axis specification, etc.?

Comment (dlwh) [reply tests/configs/misc.py:16]: done!
